code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1610,87 +1610,240 @@ GLOBAL_LIST_EMPTY(roundstart_races)
+
+		humi.adjust_bodytemperature(core_skin_change)
+
+
+/**
+ * Used to get the amount of change between two body temperatures
+ *
+ * When passed the difference between two temperatures returns the amount of change to temperature to apply.
+ * The change rate should be kept at a low value tween 0.16 and 0.02 for optimal results.
+ * vars:
+ * * temp_diff (required) The differance between two temperatures
+ * * change_rate (optional)(Default: 0.06) The rate of range multiplyer
+ */
+/datum/species/proc/get_temp_change_amount(var/temp_diff, var/change_rate = 0.06)
+	if(temp_diff < 0)
+		return (log((temp_diff * -1) * change_rate + 1) * BODYTEMP_AUTORECOVERY_DIVISOR) * -1
+	return log(temp_diff * change_rate + 1) * BODYTEMP_AUTORECOVERY_DIVISOR
+
+/**
+ * Used to set alerts and debuffs based on body temperature
+ * vars:
+ * * humi (required) The mob we will targeting
+ */
+/datum/species/proc/body_temperature_alerts(mob/living/carbon/human/humi)
 	// Body temperature is too hot, and we do not have resist traits
-	if(H.bodytemperature > bodytemp_heat_damage_limit && !HAS_TRAIT(H, TRAIT_RESISTHEAT))
+	if(humi.bodytemperature > bodytemp_heat_damage_limit && !HAS_TRAIT(humi, TRAIT_RESISTHEAT))
 		// Clear cold mood and apply hot mood
-		SEND_SIGNAL(H, COMSIG_CLEAR_MOOD_EVENT, "cold")
-		SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "hot", /datum/mood_event/hot)
+		SEND_SIGNAL(humi, COMSIG_CLEAR_MOOD_EVENT, "cold")
+		SEND_SIGNAL(humi, COMSIG_ADD_MOOD_EVENT, "hot", /datum/mood_event/hot)
 
 		//Remove any slowdown from the cold.
-		H.remove_movespeed_modifier(/datum/movespeed_modifier/cold)
+		humi.remove_movespeed_modifier(/datum/movespeed_modifier/cold)
+		// display alerts based on how hot it is
+		switch(humi.bodytemperature)
+			if(0 to 461)
+				humi.throw_alert("temp", /obj/screen/alert/hot, 1)
+			if(460 to 700)
+				humi.throw_alert("temp", /obj/screen/alert/hot, 2)
+			else
+				humi.throw_alert("temp", /obj/screen/alert/hot, 3)
 
-		var/burn_damage = 0
-		var/firemodifier = H.fire_stacks / 50
-		if (!H.on_fire) // We are not on fire, reduce the modifier
+	// Body temperature is too cold, and we do not have resist traits
+	else if(humi.bodytemperature < bodytemp_cold_damage_limit && !HAS_TRAIT(humi, TRAIT_RESISTCOLD))
+		// clear any hot moods and apply cold mood
+		SEND_SIGNAL(humi, COMSIG_CLEAR_MOOD_EVENT, "hot")
+		SEND_SIGNAL(humi, COMSIG_ADD_MOOD_EVENT, "cold", /datum/mood_event/cold)
+		// Apply cold slow down
+		humi.add_or_update_variable_movespeed_modifier(/datum/movespeed_modifier/cold, multiplicative_slowdown = ((bodytemp_cold_damage_limit - humi.bodytemperature) / COLD_SLOWDOWN_FACTOR))
+		// Display alerts based how cold it is
+		switch(humi.bodytemperature)
+			if(201 to bodytemp_cold_damage_limit)
+				humi.throw_alert("temp", /obj/screen/alert/cold, 1)
+			if(120 to 200)
+				humi.throw_alert("temp", /obj/screen/alert/cold, 2)
+			else
+				humi.throw_alert("temp", /obj/screen/alert/cold, 3)
```

Comment:
See above

Reply:
same comment, but if I can think of an improvment.